tests/src/test/java/org/sonarqube/tests/ce/CeWorkersTest.java

```diff
@@ -234,4 +330,14 @@ private static void waitInterruptedly() {
       e.printStackTrace();
     }
   }
+
+  private void waitForEmptyQueue() throws InterruptedException {
+    int tasksCount;
+    do {
+      Thread.sleep(100);
+      tasksCount = adminWsClient.ce().activity(new ActivityWsRequest()
+        .setStatus(ImmutableList.of("PENDING", "IN_PROGRESS")))
+        .getTasksCount();
+    } while (tasksCount > 0);
```

Comment:
If there is an issue in this loop the tests may keep running, adding a `org.junit.rules.Timeout`  must be added.

Reply:
I'll limit the number of loops rather than use  `org.junit.rules.Timeout` which works globally and for which no value is relevant